domain/schema/model/sql/0019-application.sql

```diff
@@ -200,6 +200,55 @@ CREATE TABLE application_status (
     REFERENCES workload_status_value (id)
 );
 
+CREATE TABLE device_constraint (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    application_uuid TEXT NOT NULL,
+    name TEXT NOT NULL,
+    type_id INT,
+    count INT,
+    attribute_uuid TEXT,
+    CONSTRAINT fk_device_constraint_application
+    FOREIGN KEY (application_uuid)
+    REFERENCES application (uuid),
+    CONSTRAINT fk_device_constraint_device_type
+    FOREIGN KEY (type_id)
+    REFERENCES device_constraint_type (id),
+    CONSTRAINT fk_device_constraint_attribute
+    FOREIGN KEY (attribute_uuid)
+    REFERENCES device_constraint_attribute (uuid)
+);
+
+CREATE TABLE device_constraint_type (
+    id INT PRIMARY KEY,
+    value TEXT
+);
+
+INSERT INTO device_constraint_type VALUES
+(0, 'gpu'),
+(1, 'nvidia.com/gpu'),
+(2, 'amd.com/gpu');
```

Comment:
Is that really the case? That's horrid.

Reply:
I've taken the values from https://github.com/juju/juju/blob/main/core/devices/constraints.go#L21-L26